test/check-machines-migrate

```diff
@@ -0,0 +1,231 @@
+        # Ports for TCP tranfer are closed
+        self._testMigrationGeneric(True, False, "port_closed")
+
+    def testFailMigrationAuthentication(self):
+        # No key authentication is set up
+        self._testMigrationGeneric(True, False, "authentication")
+
+    def testFailMigrationUriIncorrect(self):
+        # Incorrect uri inputted
+        self._testMigrationGeneric(True, False, "uri")
+
+    def testFailMigrationDomainUnknown(self):
+        # Different sized volume already exists on destination
+        self._testMigrationGeneric(True, False, "domain")
+
+    def testFailMigrationStopLibvirtd(self):
+        # Libvirtd.service is not working on the destination
+        self._testMigrationGeneric(True, False, "libvirtd")
+
+    def testFailMigrationKillDestination(self):
+        # Kill destination server in the middle of migration
+        self._testMigrationGeneric(True, False, "kill")
+
+    def _testMigrationGeneric(self, copy_storage, temporary, fail):
+        b = self.browser
+        machine1 = self.machine
+        machine2 = self.machines['machine2']
+
+        self.createVm("subVmTest1")
+        self.login_and_go("/machines")
+        self.prepare(machine1, machine2, fail)
+
+        if not copy_storage:
+            self.preCreateStorageOnDest(machine1, machine2)
+        # Older libvirt needs the destination storage preallocated
+        elif self.machine.image in ["ubuntu-2004"] and fail != "pool":
+            machine2.execute("qemu-img create -f qcow2 /var/lib/libvirt/images/subVmTest1-2.img 50M")
+
+        self.waitVmRow("subVmTest1")
+        self.goToVmPage("subVmTest1")
+
+        self.performAction("subVmTest1", "migrate")
+        b.wait_visible("#migrate-modal")
+
+        if fail == "uri":
+            # test failure for syntactically incorrect VM
+            b.set_input_text("#dest-uri-input", "qemu+ssh://root@machine2/sistem")
+        elif fail == "domain":
+            # test failure for unkwnown domain
+            b.set_input_text("#dest-uri-input", "qemu+ssh://root@someunknowndomain/system")
+        else:
+            b.set_input_text("#dest-uri-input", "qemu+ssh://root@machine2/system")
+
+        if copy_storage:
+            b.click("#copy")
+        else:
+            b.click("#shared")
+
+        b.wait_visible("#temporary")
+        b.set_checked("#temporary", temporary)
+        if temporary:
+            b.wait_visible(".pf-c-modal-box__footer .pf-c-alert")
+
+        # The VM does not exist in destination before migration happens
+        self.assertNotIn("subVmTest1", machine2.execute("virsh list --all"))
+
+        if fail == "libvirtd":
+            machine2.execute("systemctl stop libvirtd.service")
+            if machine1.image not in ["debian-stable"]:
+                machine2.execute("systemctl stop libvirtd.socket")
+
+        b.click("#migrate-button")
+        if fail:
+            if fail == "kill":
+                machine2.execute("sleep 1; kill -SEGV 1")
```

Comment:
That will oops the entire machine. Could this perhaps kill libvirt/qemu only, so that one can still ssh in and debug/watch stuff? (not a blocker, can be a follow-up, just thinking aloud)

Reply:
> That will oops the entire machine.

That was kinda a point, to simulate a catastrophic scenario, like a total server crash, or a power outage.

> debug/watch stuff

What kind of stuff should be watched? To see if VM wasn't left half-way migrated on destination?

Reply:
Right, both for automatic test inspection and for interactive debugging. But this isn't important, let's not block on this. Resolving.